modules/packages/Sort.chpl

```diff
@@ -542,34 +569,31 @@ proc quickSort(Data: [?Dom] ?eltType, minlen=16, comparator:?rec=defaultComparat
   //  cobegin {
     quickSort(Data[..loptr-1], minlen, comparator);  // could use unbounded ranges here
     quickSort(Data[loptr+1..], minlen, comparator);
-    //  }
+  //  }
 }
 
 
-// TODO -- support comparators by implementing a reduce intent w/ comparators
 /*
    Sort the 1D array `Data` in-place using a sequential selection sort
    algorithm.
 
-   .. warning::
-        SelectionSort will be deprecated for a comparator-supported
-        ``selectionSort`` in the future.
-
    :arg Data: The array to be sorted
    :type Data: [] `eltType`
-   :arg doublecheck: Verify the array is correctly sorted before returning
-   :type doublecheck: `bool`
-   :arg reverse: Sort in reverse numerical order
-   :type reverse: `bool`
+   :arg comparator: :ref:`Comparator <comparators>` record that defines how the
+      data is sorted.
 
  */
-proc SelectionSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false) where Dom.rank == 1 {
-  const lo = Dom.dim(1).low;
-  const hi = Dom.dim(1).high;
+proc selectionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator) where Dom.rank == 1 {
+  const lo = Dom.dim(1).low,
+        hi = Dom.dim(1).high;
   for i in lo..hi-1 {
-    var (_, loc) = if reverse then maxloc reduce zip(Data[i..hi], {i..hi})
-      else minloc reduce zip(Data[i..hi], {i..hi});
-    Data(i) <=> Data(loc);
+    var jMin = i;
+    // TODO -- should be a reduce intent, when they can support comparators
```

Comment:
Could you add to this comment to describe how it should do 'minloc reduce' ? The "should be a reduce intent" doesn't cover it for me.